src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.cs

```diff
@@ -448,7 +451,13 @@ internal override void OnItemsSourceSingleCollectionChanged(object sender, Notif
 					{
 						this.Log().Debug($"Inserting {args.NewItems.Count} items starting at {args.NewStartingIndex}");
 					}
+
+					// Because new items are added, the containers for existing items with higher indices
+					// will be moved, and we must make sure to increase their indices
+					SaveContainersForIndexRepair(args.NewStartingIndex, args.NewItems.Count);
```

Comment:
I'm curious, what's the reason for saving the containers, performing the add (or remove), and modifying the indices after, versus just modifying them immediately? (I'm sure there's a good reason, I just don't see immediately what it is.)

Reply:
Yes, I wanted to try that too. The reason I did it this way originally was that I noticed that some of the code deep in the `AddItems` was using some of the `IndexFromContainer`-like methods, so I didn't want to modify the indices _before_ the actual item change occurred. But it may be worth trying out whether the change has any effect. I will dive deeper into the add/remove codepaths to see if there are any risks to that